roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,159 @@
+        _image_mask = np.reshape(csc_matrix((masks, roi_ids, ids), shape=(np.prod(self.get_image_size()),self.no_rois)).toarray(),
+            [self.get_image_size()[0],self.get_image_size()[1],-1],order='F')
+        return masks, roi_ids, ids, _image_mask
+
+    def _trace_extractor_read(self, field):
+        if self._dataset_file['estimates'].get(field):
+            return self._dataset_file['estimates'][field] # lazy read dataset)
+        else:
+            return None
+
+    def _summary_image_read(self):
+        if self._dataset_file['estimates'].get('Cn'):
+            return np.array(self._dataset_file['estimates']['Cn']).T
+        else:
+            return None
+
+    def get_accepted_list(self):
+        accepted = self._dataset_file['estimates']['idx_components']
+        if len(accepted.shape)==0:
+            accepted = list(range(self.no_rois))
+        return accepted
+
+    def get_rejected_list(self):
+        rejected = self._dataset_file['estimates']['idx_components_bad']
+        if len(rejected.shape) == 0:
+            rejected = [a for a in range(self.no_rois) if a not in set(self.get_accepted_list())]
+        return rejected
+
+    @property
+    def roi_locations(self):
+        _masks, _mask_roi_ids, _mask_ids, _ = self._image_mask_sparse_read()
+        roi_location = np.ndarray([2, self.no_rois], dtype='int')
+        for i in range(self.no_rois):
+            max_mask_roi_id = _mask_roi_ids[_mask_ids[i]+np.argmax(
+                _masks[_mask_ids[i]:_mask_ids[i+1]]
+            )]
+            roi_location[:, i] = [((max_mask_roi_id+1)%(self.image_size[0]+1))-1,#assuming order='F'
+                                  ((max_mask_roi_id+1)//(self.image_size[0]+1))]
+            if roi_location[0,i]<0:
+                roi_location[0,i]=0
+        return roi_location
+
+    @staticmethod
+    def write_segmentation(segmentation_object, savepath, **kwargs):
+        plane_no = kwargs.get('plane_no', 0)
+        filename = os.path.basename(savepath)
+        savepath_folder = os.path.join(os.path.dirname(savepath), f'Plane_{plane_no}')
+        savepath = os.path.join(savepath_folder, filename)
+        if savepath.split('.')[-1]!='hdf5':
+            raise ValueError('filetype to save must be *.hdf5')
+        with h5py.File(savepath,'w') as f:
+            #create base groups:
+            estimates = f.create_group('estimates')
+            params = f.create_group('params')
+            #adding to estimates:
+            if segmentation_object._roi_response_neuropil is not None:
+                estimates.create_dataset('C',data=segmentation_object._roi_response_neuropil)
+            estimates.create_dataset('F_dff', data=segmentation_object._roi_response_fluorescence)
+            if segmentation_object._roi_response_deconvolved is not None:
+                estimates.create_dataset('S', data=segmentation_object._roi_response_deconvolved)
+            if segmentation_object._images_mean is not None:
```

Comment:
Cn is not mean image

Reply:
fixed

Reply:
It still looks to me like you are calling Cn the image_mean